### PR TITLE
Optimism Mainnet config

### PR DESCRIPTION
**Pull request by mateodaza**
## Summary by CodeRabbit

- **New Features**
  - Added `SQUID_NETWORK` environment variable to support network-specific configurations.
  - Introduced a new script `prettier:write` for formatting TypeScript files.
  
- **Bug Fixes**
  - Corrected a typo in the `assets/README.MD` file.

- **Refactor**
  - Updated project name in `package.json` from "squid-evm-template" to "devouch-squid-backend".
  - Enhanced environment-specific constants and configurations.
  - Improved import statement consistency across the project.

- **Chores**
  - Updated Prettier configuration and ignored additional directories.
  - Added conditional checks in database migration scripts based on the `SQUID_NETWORK` environment variable.

### CODE REVIEW

**Comment by mhmdksh** [comment]
@mateodaza @MohammadPCh @divine-comedian This can be a separate branch (Instead of merging to develop, as this is a different configuration (For OP) WDYT?

**Comment by mateodaza** [comment]
@mhmdksh thats perfect. I like creating PRs to easily check on changes - I agree with you! Let's do that 

You can delete this after we're done

File: src/constants.ts [hunk 1]
```diff
@@ -1,17 +1,31 @@
-import { assertNotNull } from "@subsquid/evm-processor";
+import { assertNotNull } from '@subsquid/evm-processor';
+
+const SQUID_NETWORK = process.env.SQUID_NETWORK || 'eth-sepolia';
+
+const IS_PRODUCTION = SQUID_NETWORK === 'optimism-mainnet';
```
Comment (aminlatifi) [inline src/constants.ts:5]: I am uneasy with limiting our env configuration to two production and non-production types. I think a more flexible one would be naming each env like (op-sepolia-staging, op-mainnet-production, sepolia-mainnet-staging, ...) and in the config we change only one variable. Then we can have multiple configurations with more readability.

File: .env.template [hunk 2]
```diff
@@ -14,11 +14,15 @@ RPC_ENDPOINT=https://eth-sepolia.g.alchemy.com/v2/XXXXXXXXXXXXXXXXXXXXXXXXXX
...
-PROJECT_VERIFY_ATTESTATION_SCHEMA=0x3D5854AF182F27966DEA837C446A051B3509224DDC03150E55097B362D111B1B
-PROJECT_GIVBACK_ELIGIBLE_ATTESTATION_SCHEMA=0x0000000000000000000000000000000000000000000000000000000000000000
+SQUID_NETWORK=optimism-mainnet
+
+# MOVED TO CONSTANTS
+# EAS_CONTRACT_ADDRESS=0xC2679fBD37d54388Ce493F1DB75320D236e1815e
+# SCHEMA_CONTRACT_ADDRESS=0x0a7E2Ff54e76B8E6659aedc9103FB21c038050D0
+
+# PROJECT_VERIFY_ATTESTATION_SCHEMA=0x3D5854AF182F27966DEA837C446A051B3509224DDC03150E55097B362D111B1B
+# PROJECT_GIVBACK_ELIGIBLE_ATTESTATION_SCHEMA=0x0000000000000000000000000000000000000000000000000000000000000000
```
Comment (aminlatifi) [inline .env.template:25]: I love the idea of bundling all configuration under env title, like 'op-mainnet-production',. ...

**Review by aminlatifi — CHANGES_REQUESTED** [review]
Thanks @mateodaza

File: src/constants.ts (hunk 1 above)
Comment (aminlatifi) [reply src/constants.ts:5]: Let's stick to this for fast follow

**Review by aminlatifi — APPROVED** [review]
Thanks @mateodaza , GJ
I did some updates in the migration generation to support multi env (production and staging)
Plus integrated with prettier, if you prefer single comma (me too), we can change the configuration and run `prettier:write` script. But I prefer to do it in another PR.
Also, I did a bug fix in reading START_BLOCK config.

**Comment by aminlatifi** [comment]
@mateodaza I tested the op mainnet env on my machine and it was ok.

**Pull request merged**